src/main/java/org/wso2/carbon/module/swiftiso20022/utils/JsonToMt940Utils.java

```diff
@@ -0,0 +1,326 @@
+     * @param requestPayload   Request Payload
+     * @return  Whether the sequence number is valid or not
+     */
+    public static boolean validateSequenceNumber(RequestPayloadModel requestPayload) {
+        return (requestPayload.getSequenceNumber() != null && !requestPayload.getSequenceNumber().isBlank() &&
+                requestPayload.getSequenceNumber().length() < 4);
+    }
+
+    /**
+     * Method to validate the Balance Details in the request payload.
+     *
+     * @param balanceDetails   Balance Details
+     * @param fieldName        Balance Field Name
+     * @return  Return ErrorModel if the balance is invalid.
+     */
+    public static ErrorModel validateBalanceDetails(BalanceModel balanceDetails, String fieldName) {
+
+        if (!isValidDateFormat(balanceDetails.getDate())) {
+            return new ErrorModel(ConnectorConstants.ERROR_T50,
+                    String.format(ConnectorConstants.ERROR_BAL_DATE_INVALID, fieldName));
+        }
+
+        if (!ValidatorUtils.isValidCurrency(balanceDetails.getCurrency())) {
+            return new ErrorModel(ConnectorConstants.ERROR_T52,
+                    String.format(ConnectorConstants.ERROR_CURRENCY_CODE_INVALID,
+                            ConnectorConstants.OPENING_BALANCE));
+        }
+
+        ErrorModel error = ValidatorUtils.validateAmountLength(balanceDetails.getBalanceAmount(),
+                fieldName);
+        if (error.isError()) {
+            return error;
+        }
+
+        if (balanceDetails.getIndicator() == null || balanceDetails.getIndicator().isBlank() ||
+                !(ConnectorConstants.DEBIT.equals(balanceDetails.getIndicator()) ||
+                ConnectorConstants.CREDIT.equals(balanceDetails.getIndicator()))) {
+            return new ErrorModel(ConnectorConstants.ERROR_T51, ConnectorConstants.ERROR_BAL_IND_INVALID);
+        }
+
+        if (balanceDetails.getStatementType() != null && !balanceDetails.getStatementType().isBlank()) {
+            if (!(ConnectorConstants.CURRENT_STATEMENT_TYPE.equals(balanceDetails.getStatementType()) ||
+                    ConnectorConstants.LAST_STATEMENT_TYPE.equals(balanceDetails.getStatementType()))) {
+                return new ErrorModel(ConnectorConstants.ERROR_T51, ConnectorConstants.ERROR_INVALID_STATEMENT_TYPE);
+            }
+        }
+
+        return new ErrorModel();
+    }
+
+    /**
+     * Method to validate the date format.
+     *
+     * @param dateTime  Date time to validate
+     * @return          True if the date format is valid, else false
+     */
+    public static boolean isValidDateFormat(String dateTime) {
+
+        if (dateTime == null || dateTime.isBlank()) {
+            return false;
+        }
+        DateFormat formatter = new SimpleDateFormat(ConnectorConstants.DATE_TIME_FORMAT);
+
+        try {
+            formatter.parse(dateTime);
+            return true;
+        } catch (ParseException e) {
+            return false;
```

Comment:
Shall we add a debug log here?

Reply:
Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/2/commits/8d6b87e8988955fbe9679e7afa9d4285b0ff0ac8